src/docfx/Errors.cs

```diff
@@ -392,6 +392,9 @@ public static Error AtXrefNotFound(SourceInfo<string> source)
             public static Error XrefNotFound(SourceInfo<string> source)
                 => new Error(ErrorLevel.Warning, "xref-not-found", $"Cross reference not found: '{source}'.", source);
 
+            public static Error InvalidXrefType(SourceInfo<string> xref, string expectedXrefType, string? realXrefType)
+               => new Error(ErrorLevel.Warning, "xref-invalid-type", $"Cross reference '{xref}' invalid type, '{expectedXrefType}' is expected, but now is '{realXrefType}'.", xref);
```

Comment:
Change the error code to `xref-type-invalid` to align with other error code style, and also change the message to match `unexpected-type` (which has been reviewed)
```suggestion
               => new Error(ErrorLevel.Warning, "xref-type-invalid", $"Invalid cross reference: '{xref}'. Expected type '{expectedXrefType}' but got '{actualXrefType}'.", xref);
```

Reply:
ok.